src/skottie-player.ts

```diff
@@ -753,6 +793,131 @@ export class SkottiePlayer extends LitElement {
   }
 
+  /**
+   * Repositions text to not cover faces.
+   */
+  repositionText(): void {
+    const oldCanvas = document.getElementById('canvas');
+    if (oldCanvas) {
+      oldCanvas.remove();
+    }
+    const fileInput = (this.shadowRoot?.getElementById('img_1.png-input') as HTMLInputElement);
+    const file = fileInput.files?.[0];
+    if (file) {
+      createImageBitmap(file).then((bitmap) => {
+        const canvasElement = document.createElement('canvas');
+        canvasElement.setAttribute('id', 'canvas');
+        canvasElement.setAttribute('width', this.width.toString());
+        canvasElement.setAttribute('height', this.height.toString());
+        const context = canvasElement.getContext('2d');
+        if (context) {
+          context.drawImage(bitmap, 0, 0, bitmap.width, bitmap.height,
+                            0, 0, this.width, this.height);
+          this.faceBounder.getFaceBounds(canvasElement).then((result) => {
+            console.log(result)
+            const [xValues, yValues, faceArray] = this.getXYValues(result);
+            console.log(faceArray);
+            
+            const [topLeft, bottomRight] = this.getTextBounds(xValues, yValues, faceArray);
+            console.log(topLeft, bottomRight);
+
+            if (this.content.assets) {
+              this.content.assets[9].layers[0].t.d.k[0].s.sz[0] = bottomRight[0] - topLeft[0];
+              this.content.assets[9].layers[0].t.d.k[0].s.sz[1] = bottomRight[1] - topLeft[1];
+              this.content.assets[9].layers[0].t.d.k[0].s.ps[0] = topLeft[0] - this.width / 2;
+              this.content.assets[9].layers[0].t.d.k[0].s.ps[1] = topLeft[1] - this.height / 2;
+            }
+
+            this.updateJson();
+          });
+        }
+      })
+    }
+  }
+
+  /**
+   * Gets potential x and y coordinates for the text bounding box.
+   */
+  private getXYValues(faceArray: Array<any>): Array<Array<number>> {
+    const margin = 32;
+    let xValues = [margin, this.width - margin];
+    let yValues = [margin, this.height - margin];
+    for (let i = 0; i < faceArray.length; i++) {
+      faceArray[i].topLeft[0] -= margin;
+      faceArray[i].topLeft[1] -= margin;
+      faceArray[i].bottomRight[0] += margin;
+      faceArray[i].bottomRight[1] += margin;
+      xValues.push(faceArray[i].topLeft[0]);
+      yValues.push(faceArray[i].topLeft[1]);
+      xValues.push(faceArray[i].bottomRight[0]);
+      yValues.push(faceArray[i].bottomRight[1]);
+    }
+    return [xValues, yValues, faceArray];
+  }
+
+  /**
+   * Gets the largest text box possible that doesn't overlap with any of the
+   * faces in the face array.
+   */
+  private getTextBounds(xValues: Array<number>, yValues: Array<number>,
+    faceArray: Array<any>): Array<Array<number>> {
+    let topLeft = [xValues[0], yValues[0]];
+    let bottomRight = [xValues[1], yValues[1]];
+    let maxArea = 0;
+    for (let x1 = 0; x1 < xValues.length; x1++) {
```

Comment:
x1++

Should this be x1+=2, because I think x1 and x2 are intervened inside xValues?